pkg/apis/provisioning/v0alpha1/types.go

```diff
@@ -68,36 +68,71 @@ const (
 )
 
 type RepositorySpec struct {
-	// Describe the feature toggle
+	// The repository display name (shown in the UI)
 	Title string `json:"title"`
 
-	// Describe the feature toggle
+	// Repository description
 	Description string `json:"description,omitempty"`
 
-	// The folder that is backed by the repository.
-	// The value is a reference to the Kubernetes metadata name of the folder in the same namespace.
-	Folder string `json:"folder,omitempty"`
-
 	// Edit options within the repository
 	Editing EditingOptions `json:"editing"`
 
+	// Sync settings -- how values are pulled from the repository into grafana
+	Sync SyncOptions `json:"sync"`
+
 	// The repository type.  When selected oneOf the values below should be non-nil
 	Type RepositoryType `json:"type"`
 
 	// The repository on the local file system.
-	// Mutually exclusive with s3 and github.
+	// Mutually exclusive with local | s3 | github.
 	Local *LocalRepositoryConfig `json:"local,omitempty"`
 
 	// The repository in an S3 bucket.
-	// Mutually exclusive with local and github.
+	// Mutually exclusive with local | s3 | github.
 	S3 *S3RepositoryConfig `json:"s3,omitempty"`
 
 	// The repository on GitHub.
-	// Mutually exclusive with local and s3.
+	// Mutually exclusive with local | s3 | github.
 	// TODO: github or just 'git'??
 	GitHub *GitHubRepositoryConfig `json:"github,omitempty"`
 }
 
+// SyncTargetType defines where we want all values to resolve
+// +enum
+type SyncTargetType string
+
+// RepositoryType values
+const (
+	// Resources are saved in the global context
+	// Only one repository may specify the `instance` target
+	// When this exists, the UI will promote writing to the instance repo
+	// rather than the grafana database (where possible)
+	SyncTargetTypeInstance SyncTargetType = "instance"
+
+	// Resources will be saved into a folder managed by this repository
+	// The folder k8s name will be the same as the repository k8s name
+	// It will contain a copy of everything from the remote
+	SyncTargetTypeMirror SyncTargetType = "mirror"
+
+	// Pick an explicit folder where we can save resources, not managed by provisioning
+	// SyncTargetTypeFolder SyncTargetType = "folder"
+)
+
+type SyncOptions struct {
+	// Enabled must be saved as true before any sync job will run
+	Enabled bool `json:"enabled"`
+
+	// Where values should be saved
```

Comment:
yes, a "mirror" target for the folder setup doesn't sound right -> follow up, not needed to address here.